### PR TITLE
[GEOT-6961] - WFSContentComplexFeatureSource.getFeatures must use correct namespace

### PR DESCRIPTION
**Pull request by roarbra**
[GEOT-6961](https://osgeo-org.atlassian.net/browse/GEOT-6961) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

There were some problems calling getFeatures for a WFS with complex structure. With these changes, I managed to iterate through the whole response.

In addition to the problems described in the ticket, there was a problem with reading geometry. That I solved in the same manner as other complex structures within that function.
  

# Checklist

- [X] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [X] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [X] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [X] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [X] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [X] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by roarbra** [comment]
I see this is an unsupported module, but these things got very stagnant.
That's a pity because I got more to come, and this PR is blocking my additional work.

@aaime Should I just close this one, and start over with a new PR?

**Comment by aaime** [comment]
This week with FOSS4G going everyone is pretty busy... patience.

File: modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/WFSContentComplexFeatureCollection.java [hunk 1]
```diff
@@ -74,12 +74,13 @@ public WFSContentComplexFeatureCollection(
...
             InputStream stream = request.getFinalURL().openStream();
             XmlComplexFeatureParser parser =
-                    new XmlComplexFeatureParser(stream, schema, name, filter);
+                    new XmlComplexFeatureParser(
+                            stream, schema, name, filter, request.getStrategy());
             return new ComplexFeatureIteratorImpl(parser);
         } catch (IOException e) {
-            LOGGER.log(Level.FINER, e.getMessage(), e);
+            LOGGER.log(Level.SEVERE, e.getMessage(), e);
+            throw new RuntimeException("Couldn't read features of collection.", e);
```
Comment (nmco) [inline modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/WFSContentComplexFeatureCollection.java:82]: Nice.

**Comment by nmco** [comment]
@roarbra looks good to me, that says @roarbra complex features are ... well quite complex, hence before merging a PR that touches them is a good practice to run the App-Schema integration tests.

The full battery of integration tests is implemented on the GeoServer side, [HERE](https://github.com/geoserver/geoserver/tree/main/src/extension/app-schema). Did you ha da chance to run them?

**Comment by roarbra** [comment]
I haven't looked at app-schema. I could do so.

**Comment by aaime** [comment]
@nmco I have the impression the changes here are all contained in the wfs-ng module, no core app-schema class has been touched. @roarbra can you confirm?

**Comment by roarbra** [comment]
Yes, my changes are in gt-wfs-ng only.

**Comment by aaime** [comment]
Oh well, let's get this in, don't think there is much usage for wfs + complex features anyways...

**Pull request merged**